### PR TITLE
refactor(typescript): noImplicitAny for message-stream test file

### PR DESCRIPTION
**Pull request by praveenqlogic**
towards #313 (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: test/message-stream.ts [hunk 1]
```diff
@@ -142,7 +144,7 @@ describe('MessageStream', () => {
     const gaxClient = new FakeGaxClient();
     client = gaxClient.client;  // we hit the grpc client directly
     subscriber = new FakeSubscriber(gaxClient);
-    messageStream = new MessageStream(subscriber);
+    messageStream = new MessageStream(subscriber as {} as Subscriber);
```
Comment (callmehiphop) [inline test/message-stream.ts]: I'm seeing this cast all over the test file. Why not just cast the subscriber in the line above?

File: test/message-stream.ts [hunk 2]
```diff
@@ -156,20 +158,21 @@ describe('MessageStream', () => {
         objectMode: true,
         highWaterMark: 0,
       };
-
-      assert.deepStrictEqual(messageStream.options, expectedOptions);
+      // tslint:disable-next-line no-any
+      assert.deepStrictEqual((messageStream as any).options, expectedOptions);
```
Comment (callmehiphop) [inline test/message-stream.ts]: If you cast this as `FakePassThrough` you can avoid the any cast/comment.

File: test/message-stream.ts [hunk 3]
```diff
@@ -156,20 +158,21 @@ describe('MessageStream', () => {
...
+          new MessageStream(subscriber as {} as Subscriber, {highWaterMark});
 
       const expectedOptions = {
         objectMode: true,
         highWaterMark,
       };
-
-      assert.deepStrictEqual(ms.options, expectedOptions);
+      // tslint:disable-next-line no-any
+      assert.deepStrictEqual((ms as any).options, expectedOptions);
```
Comment (callmehiphop) [inline test/message-stream.ts]: If you cast this as `FakePassThrough` you can avoid the any cast/comment.

File: test/message-stream.ts [hunk 4]
```diff
@@ -338,7 +344,7 @@ describe('MessageStream', () => {
         const fakeResponses = [{}, {}, {}, {}, {}];
         const recieved: object[] = [];
 
-        messageStream.on('data', chunk => recieved.push(chunk))
+        messageStream.on('data', chunk => recieved.push(chunk as object))
```
Comment (callmehiphop) [inline test/message-stream.ts]: Seems like if we can avoid casting emitted data we should. Does implementing an interface that lists the correct types work here?

```typescript
interface StreamEvents {
  on(event: 'data', listener: (data: {}) => void): void;
}

class MessageStream implements StreamEvents {}
```

?

File: test/message-stream.ts (hunk 1 above)
Comment (praveenqlogic) [reply test/message-stream.ts]: Done

File: test/message-stream.ts (hunk 3 above)
Comment (praveenqlogic) [reply test/message-stream.ts]: done thanks 👍

**Pull request merged**